pkg/v1/partial/uncompressed.go

```diff
@@ -207,6 +218,48 @@ func (i *uncompressedImageExtender) Layers() ([]v1.Layer, error) {
 	return ls, nil
 }
 
+// CompressedLayers implements v1.Image
+// This function compresses layers in parallel, so it is fairly expensive computationally.
+func (i *uncompressedImageExtender) CompressedLayers() ([]v1.Layer, error) {
+	if i.compressedLayers != nil {
+		return i.compressedLayers, nil
+	}
+
+	diffIDs, err := DiffIDs(i)
+	if err != nil {
+		return nil, err
+	}
+
+	ls := make([]v1.Layer, len(diffIDs), len(diffIDs))
+
+	var g errgroup.Group
+	for idx, d := range diffIDs {
+		idx := idx
+		d := d
+
+		g.Go(func() error {
+			l, err := i.LayerByDiffID(d)
+			if err != nil {
+				return err
+			}
+
+			ls[idx], err = ToCompressedLayer(l)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
```

Comment:
Parallelizing here saved >1 minute.

Reply:
I think we might actually want to move the parallelization into here?

https://github.com/google/go-containerregistry/blob/b02d448a3705facf11018efff34f1d2830be5724/pkg/v1/mutate/mutate.go#L153-L190

Reply:
Do uncompressed images back on to `mutate`? I haven't traced it deep enough into the stack.